### PR TITLE
DOC: fix typo and improve cmake instructions

### PR DESCRIPTION
**Pull request by ssardina**
- Fix small typos and fonts
- Improve documentation of how to meet pre-reqs before building
- Add more explanation on running SWIPL when having multiple installs (distribution, local install, build)

### CODE REVIEW

**Comment by JanWielemaker** [comment]
Thanks.   Merged after adding a test for the conflicting variables to `swipl-activate` and print a warning from there rather than documenting the issue.

File: CMAKE.md [hunk 1]
```diff
@@ -52,10 +57,7 @@ latest version:
     git pull
     git submodule update --init
```
Comment (kamahen) [inline CMAKE.md:58]: Isn't `git submodule  update --init` rather out of date? Instead: `git pull --recurse=yes`? (The `git clone` has the recursive option)

File: CMAKE.md [hunk 2]
```diff
@@ -52,10 +57,7 @@ latest version:
...
     cd build
-	cmake .
-    ninja
-    ctest -j 8
-    ninja install
+	cmake .. ninja ctest -j 8 ninja install
```
Comment (kamahen) [inline CMAKE.md:60]: I did:
```
cd ~/src/swipl-devel/build && cmake .. ninja ctest -j 8 ninja install
```
and got:
```
CMake Warning:
  Ignoring extra path from command line:

   "install"


CMake Error: Unknown argument -j
CMake Error: Run 'cmake --help' for all supported options.
```

File: CMAKE.md (hunk 1 above)
Comment (ssardina) [reply CMAKE.md:58]: Good point, yes the recurse option has been added to most git commands, including `pull`. 

Presumably it is `--recurse-submodules=yes` or simply `--recurse-submodules`, but seems `--recurse`  also works!

File: CMAKE.md (hunk 2 above)
Comment (ssardina) [reply CMAKE.md:60]: @kamahen , but that does not seem to be the right command. Isn´t that meant to be:

```
cd ~/src/swipl-devel/build && cmake -G ninja .. && ninja && ctest -j 8 && ninja install
```

**Comment by ssardina** [comment]
> Thanks. Merged after adding a test for the conflicting variables to `swipl-activate` and print a warning from there rather than documenting the issue.

Fantastic, much better handling with a better script now. :clap:

File: CMAKE.md (hunk 2 above)
Comment (kamahen) [reply CMAKE.md:60]: (Yes - I forgot the `cmake -G Ninja ..` because I have the commands in a bash script with `-e`)

Comment (JanWielemaker) [reply CMAKE.md:60]: Apparently @ssardina's edits changes the multi-line command into a single line.  I changed that back in the final edit, so all should be fine with the file as it is now.

Comment (ssardina) [reply CMAKE.md:60]: ouch! sorry about that, that's sloppy. But it is strange, I didn't mean to do that and I dont see see that glitch on my side:

image

need to keep an eye on this, quite confused.

**Pull request closed without merging**